src/net.cpp

```diff
@@ -930,6 +938,15 @@ bool CConnman::AttemptToEvictConnection()
     for (CNode* pnode : m_nodes) {
         if (pnode->GetId() == *node_id_to_evict) {
             LogPrint(BCLog::NET, "selected %s connection for eviction peer=%d; disconnecting\n", pnode->ConnectionTypeAsString(), pnode->GetId());
+
+            TRACE6(net, evicted_connection,
+                pnode->GetId(),
+                pnode->m_addr_name.c_str(),
+                pnode->ConnectionTypeAsString().c_str(),
+                pnode->ConnectedThroughNetwork(),
+                pnode->nKeyedNetGroup,
+                std::chrono::duration<uint64_t>(pnode->m_connected).count());
```

Comment:
Thinking a little bit about what data should go into the tracepoints.

`m_addr_name`: would it make more sense to put `addr` in binary form (or both)? I'm not sure they are equivalent all the time.
`nKeyedNetGroup`: is derived deterministically from `addr` using random seeds that are created with every `CConnman`. I think that means that across restarts of a node previous net groups loose meaning (i.e. you can't compare them to netgroups of new connections). Maybe `NetGroupManager::GetGroup(addr)` would be better?

Reply:
I can imagine some use cases where having consistent netgroup identifiers would be advantageous. Unless `NetGroupManager::GetGroup()` adds too much overhead, they seem preferable.

Reply:
Looking into `GetGroup()` I saw it returns a C++ `std::vector<unsigned char>`. We can't pass C++ `std::vector`'s in the tracepoints as the eBPF code, written in C, won't be able to read it. We can call `data()` and pass this in the tracepoint as pointer to unsigned chars (we do this with e.g. the raw P2P messages too). To read from this pointer, we also need to pass the of the length of underlying array.

I'll give this a shot in a separate branch.

Reply:
An alternative is to pass `GetMappedAS()` which is an `uint32_t` that will be !=0 in an ASMap future. A net group similar to Bitcoin Core's can be calculated from the peer address by consumers that need it.